codeflash/optimization/function_optimizer.py

```diff
@@ -702,25 +713,41 @@ def reformat_code_and_helpers(
         return new_code, new_helper_code
 
     def replace_function_and_helpers_with_optimized_code(
-        self, code_context: CodeOptimizationContext, optimized_code: str, original_helper_code: str
+        self, code_context: CodeOptimizationContext, optimized_code: CodeStringsMarkdown, original_helper_code: str
     ) -> bool:
         did_update = False
         read_writable_functions_by_file_path = defaultdict(set)
         read_writable_functions_by_file_path[self.function_to_optimize.file_path].add(
             self.function_to_optimize.qualified_name
         )
+
+        file_to_code_context = optimized_code.file_to_path()
+
         for helper_function in code_context.helper_functions:
             if helper_function.jedi_definition.type != "class":
                 read_writable_functions_by_file_path[helper_function.file_path].add(helper_function.qualified_name)
+
         for module_abspath, qualified_names in read_writable_functions_by_file_path.items():
+            relative_module_path = str(module_abspath.relative_to(self.project_root))
+            logger.debug(f"applying optimized code to: {relative_module_path}")
+
+            scoped_optimized_code = file_to_code_context.get(relative_module_path)
+            if scoped_optimized_code is None:
+                logger.warning(
```

Comment:
i wonder if this code can be moved to the definition of `replace_function_definitions_in_module`. this will make this function simpler and make calling the `replace_function_definitions_in_module` safer